### DISCUSSION TITLE
How to `sudo echo 3 out> /proc/sys/vm/drop_caches` ?

### DISCUSSION THREAD

**Discussion by steelywing** [nushell/nushell#11985] [Q&A] [answered]
`sudo echo 3 out> /proc/sys/vm/drop_caches` says **Permission denied**

**Comment by fdncred** [comment, marked as answer]
when you do `sudo` nushell believes that's an external command, so it runs it like any external process, which doesn't have access to nushell. i think we need a dedicated sudo command within nushell to work properly. no one has contributed that yet.

this is what i came up with. i think it works but not positive.
```nushell
sudo /path/to/nu -c 'echo 3 out> /proc/sys/vm/drop_caches'
```

**Reply by Taywee** [reply]
Yeah, you can't sudo redirect output in other shells either, and you need the same kind of solution.

A nicer solution that I've always used for this kind of thing is identical in nu to other shells: `echo 3 | sudo tee /proc/sys/vm/drop_caches`.
